server/stream.go

```diff
@@ -3313,10 +3313,21 @@ func (mset *stream) getConsumers() []*consumer {
 	return obs
 }
 
+// Lock should be held for this one.
+func (mset *stream) numPublicConsumers() int {
```

Comment:
Looks like you are keeping track of the directs in setConsumer() and remoteConsumer(), so would not it be simpler here to do `len(mset.consumers) - mset.direct`?